src/Compilers/VisualBasic/Portable/Semantics/TypeInference/TypeInferenceCollection.vb

```diff
@@ -577,11 +579,11 @@ Namespace Microsoft.CodeAnalysis.VisualBasic
                 For Each competitor As DominantTypeData In Me.GetTypeDataList()
 
                     ' Do not merge array literals with other expressions
-                    If TypeOf competitor.ResultType IsNot ArrayLiteralTypeSymbol AndAlso type.IsSameTypeIgnoringAll(competitor.ResultType) Then
-
+                    If TypeOf competitor.ResultType IsNot ArrayLiteralTypeSymbol AndAlso type.IsSameType(competitor.ResultType, TypeCompareKind.AllIgnoreOptionsForVB) Then
```

Comment:
It's the same. I'll revert. 
This made more sense when I had a comparison for tuple names right below (now it is inside MergeTupleNames).